### ISSUE TITLE
<some_video_id> This request was detected as a bot. Use `use_po_token=True` to view

### ISSUE THREAD

**Issue by GerrySant** [cleong110/explore-youtube-asl#1]
I have been trying to download the videos and captions from Youtube following the README.md steps. After using python `download_vids.py --dataset_folder “downloads” --download_captions`, everything works fine until suddenly I get errors like `<some_video_id>:    <some_video_id> This request was detected as a bot. Use `use_po_token=True` to view`.

I have checked that this is not a direct error of your repository, but it also happens with `pytubefix`, but I was wondering how you managed to avoid the error.

**Comment by cleong110** [comment]
I also encountered this! I noticed a few things:
1. It is a bit inconsistent, not triggering on every video every time. 
2. It seems to be dependent on how quickly you are making requests. 

I therefore updated my script to (1) simply wait a random amount of time between each batch and (2) retry downloading the videos with errors. 

Took me a few days but I successfully did it. 

I also noticed that the bot detection seemed to trigger less consistently when I was doing it from a computer at a university, perhaps because YouTube receives many more requests from that IP?

**Comment by cleong110** [comment]
Seems this is caused by the latest bot detection thing. Updated the script again so as to incorporate the po_token mechanism. 

See  https://github.com/JuanBindez/pytubefix/issues/226 for details.

tl;dr you 1. generate a token, 2, run the download_vids.py script with the new flag `--use_po_token` (on by default), and 3. copy-paste the token in when it asks.

Generating a po_token: run https://colab.research.google.com/drive/190BC9tEc0wTtCsfSbWHY-lT6amjSwFyG?usp=sharing, which is basically just this:
```
npm install youtube-po-token-generator
git clone https://github.com/YunzheZJU/youtube-po-token-generator.git
node /content/youtube-po-token-generator/examples/one-shot.js
```

**Comment by cleong110** [comment]
Using the steps above, I have managed to download a large chunk of the YouTube-SL-25 dataset with no bot errors.

**Comment by indiacr** [comment]
Can this method be used to make an ASL dataset of around 10000.

**Comment by cleong110** [comment]
@indiacr What do you mean make an ASL dataset? 

If by "make a dataset" you mean carefully making a list of videos containing ASL, then that has already been done by YouTube-ASL, you can read about it in the paper: 
* YouTube ASL: https://proceedings.neurips.cc/paper_files/paper/2023/hash/5c61452daca5f0c260e683b317d13a3f-Abstract-Datasets_and_Benchmarks.html
* YouTube-Sl-25 : https://arxiv.org/abs/2407.11144

If you mean by "download 10,000 videos from the existing list", then yes, that is what the code is intended to do. I downloaded about 10k videos from the list.
